test/unit-test/sigv4_utest.c

```diff
@@ -77,6 +77,7 @@
 #define HEADERS                                               "Host: iam.amazonaws.com\r\nContent-Type: application/x-www-form-urlencoded; charset=utf-8\r\nX-Amz-Date: "DATE "\r\n\r\n"
 #define HEADERS_LENGTH                                        ( sizeof( HEADERS ) - 1U )
 #define PRECANON_HEADER                                       "content-type:application/json;\nhost:iam.amazonaws.com\n"
+#define HEADERS_WITH_X_AMZ_CONTENT_SHA256                     "Host: iam.amazonaws.com\r\nContent-Type: application/x-www-form-urlencoded; charset=utf-8\r\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b785\r\nX-Amz-Date: "DATE "\r\n\r\n"
```

Comment:
This isn't a valid hash. I'd expect the test to fail in this case instead of returning success
```suggestion
#define HEADERS_WITH_X_AMZ_CONTENT_SHA256                     "Host: iam.amazonaws.com\r\nContent-Type: application/x-www-form-urlencoded; charset=utf-8\r\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\r\nX-Amz-Date: "DATE "\r\n\r\n"
```

Reply:
We are checking the coverage here and not checking whether the signature generated matches the one generated at AWS end too.

Reply:
We should still avoid using invalid parameters in tests when possible. If the implementation is changed to verify the length of the hash, then this test will start failing for a reason unrelated to what it's meant to test for